bazel: update toolchains to support remote execution

rickystewart opened the pull request:
We need to add some extra files to `compiler_files` and `linker_files` and update some compiler flags to make sure remote execution works.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/100313)
<!-- Reviewable:end -->

rickystewart committed:
bazel: update toolchains to support remote execution

We need to add some extra files to compiler_files and linker_files and
update some compiler flags to make sure remote execution works.

Epic: none
Release note: None

rickystewart commented:
bors r=rail
